carma_wm/src/WMListenerWorker.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <lanelet2_extension/utility/message_conversion.h>
+#include "WMListenerWorker.h"
+
+namespace carma_wm
+{
+
+  WMListenerWorker::WMListenerWorker() {
+    world_model_.reset(new CARMAWorldModel);
+  }
+
+  WorldModelConstPtr WMListenerWorker::getWorldModel() const {
+    return std::static_pointer_cast< const WorldModel >(world_model_); // Cast pointer to const variant
+  }
+
+  void WMListenerWorker::mapCallback(autoware_lanelet2_msgs::MapBin map_msg) {
+
+    lanelet::LaneletMapPtr new_map(new lanelet::LaneletMap);
+
+    lanelet::utils::conversion::fromBinMsg(map_msg, new_map);
+
+    world_model_->setMap(new_map);
+
+    // Call user defined map callback
+    if (map_callback_) {
```

Comment:
Does an empty `std::function` object evaluate to false? Not sure what the initialization behaviors are for it, but it shouldn't be nullptr considering it's a class member, and would thus have it's default constructor invoked.

Reply:
It does